From ac192fc5c44cf19a04238634eafb58a38ab4c782 Mon Sep 17 00:00:00 2001
From: yvanliang <yvan6liang@outlook.com>
Date: Fri, 9 Oct 2026 15:42:51 +0800
Subject: [PATCH 05/10] GSI Android 17 compatibility:
 0006-When-logging-UiccSlot-ignore-error-if-we-cant-print-.patch

Change-Id: If9ad4ec33761df2420e940bdf2cb76e7f8a51f03
---
 .../com/android/internal/telephony/uicc/UiccSlot.java     | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/src/java/com/android/internal/telephony/uicc/UiccSlot.java b/src/java/com/android/internal/telephony/uicc/UiccSlot.java
index 700dd5f..bfc5438 100644
--- a/src/java/com/android/internal/telephony/uicc/UiccSlot.java
+++ b/src/java/com/android/internal/telephony/uicc/UiccSlot.java
@@ -822,8 +822,14 @@ public class UiccSlot extends Handler {
     @NonNull
     @Override
     public String toString() {
+        Map<Integer,String> printableIccIds = null;
+        try {
+            printableIccIds = getPrintableIccIds();
+        } catch(Throwable t) {
+            android.util.Log.e("PHH", "getPrintableIccIds failed", t);
+        }
         StringBuilder sb = new StringBuilder("[UiccSlot: mActive=" + mActive
-                + ", mIccId=" + getPrintableIccIds() + ", mIsEuicc=" + mIsEuicc
+                + ", mIccId=" + printableIccIds + ", mIsEuicc=" + mIsEuicc
                 + ", MEP=" + isMultipleEnabledProfileSupported() + ", mPortIdxToPhoneId="
                 + mPortIdxToPhoneId + ", mEid=" + Rlog.pii(TelephonyUtils.IS_DEBUGGABLE, mEid)
                 + ", mCardState=" + mCardState + " mSupportedMepMode=" + mSupportedMepMode);
